app/services/order_invoice_comparator.rb

```diff
@@ -0,0 +1,46 @@
+class OrderInvoiceComparator
+  def equal?(invoice1,invoice2)
+      # We'll use non-recursive version of the BFS algorithm to find if current state of invoice is outdated
+      # the root will be the order
+      # On each node, we'll a list of relevant attributes that will be used on the comparison
+
+      tree1 = invoice1.presenter
+      tree2 = invoice2.presenter
+      queue1 = [tree1]
+      queue2 = [tree2]
+
+      while  !queue1.empty? && !queue2.empty?
```

Comment:
This means if one queue has additional objects but they are all `nil` then we compare them with the `nil` result of `shift`ing an empty array. Those trees are seen as equal. Is that intended?

Reply:
This will disappear on the recursive version.